editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -167,11 +167,28 @@ async fn update(
 ) -> Result<Json<RollingStockWithLiveries>> {
     let data = data.into_inner();
     let rolling_stock_id = path.into_inner();
-    assert_rolling_stock_unlocked(
-        retrieve_existing_rolling_stock(&db_pool, rolling_stock_id).await?,
-    )?;
-
-    let rolling_stock = data.into_rolling_stock_model(rolling_stock_id);
+    let existing_rolling_stock =
+        retrieve_existing_rolling_stock(&db_pool, rolling_stock_id).await?;
+    assert_rolling_stock_unlocked(existing_rolling_stock.clone())?;
+    let version = existing_rolling_stock.version;
+
+    let mut rolling_stock = data.into_rolling_stock_model(rolling_stock_id);
+
+    let name_changed = existing_rolling_stock.name != rolling_stock.name;
+    let metadata_changed = existing_rolling_stock.metadata != rolling_stock.metadata;
+    let other_fields_changed = existing_rolling_stock != rolling_stock;
+
+    if name_changed || metadata_changed {
+        if other_fields_changed {
+            rolling_stock.version = version.map(|v| v + 1);
+        } else {
+            rolling_stock.version = existing_rolling_stock.version;
+        }
+    } else if other_fields_changed {
+        rolling_stock.version = version.map(|v| v + 1);
+    } else {
+        rolling_stock.version = existing_rolling_stock.version;
+    }
```

Comment:
It feels like the ` if name_changed || metadata_changed {` condition doesn't do much. Shouldn't this be:
```suggestion
    if other_fields_changed {
        rolling_stock.version = version.map(|v| v + 1);
    } else {
        rolling_stock.version = existing_rolling_stock.version;
    }
```
If it can be simplified, then you wouldn't need the `*_change` variables